### PR TITLE
Publish image factory scripts + configuration publically for customers

### PR DESCRIPTION
**Pull request by petehauge**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@petehauge__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: Scripts/ImageFactory/Scripts/DistributeImages.ps1 [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    [int] $maxConcurrentJobs
+
+)
+
+#resolve any relative paths in ConfigurationLocation 
+$ConfigurationLocation = (Resolve-Path $ConfigurationLocation).Path
+
+$scriptFolder = Split-Path $Script:MyInvocation.MyCommand.Path
+$ModulePath = Join-Path $scriptFolder "DistributionHelpers.psm1"
+Import-Module $ModulePath
```
Comment (mmydland) [inline Scripts/ImageFactory/Scripts/DistributeImages.ps1]: $modulePath?

File: Scripts/ImageFactory/Scripts/GenerateImageList.ps1 [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                </tr>
+            </thead>
+"@
+
+# resolve path for image definitions
+$imageListLocation = (Resolve-Path $ImageDefinitionsLocation).Path
+
+# get a list of artifacts & load up names/descriptions into a hashtable for reference later
+$artifactDescriptions = @{}
+$ALList = $ArtifactLocations | ForEach-Object {$_ | Resolve-Path | Get-ChildItem -Recurse -Filter "ArtifactFile.json"}
```
Comment (mmydland) [inline Scripts/ImageFactory/Scripts/GenerateImageList.ps1]: consider camelCase for ALList  and perhaps a better name for $AL in the foreach

File: Scripts/ImageFactory/Scripts/MakeGoldenImageVMs.ps1 [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        }
+    }
+
+    return $retval
+}
+
+#resolve any relative paths in ConfigurationLocation 
+$ConfigurationLocation = (Resolve-Path $ConfigurationLocation).Path
+
+$ModulePath = Join-Path (Split-Path ($Script:MyInvocation.MyCommand.Path)) "DistributionHelpers.psm1"
```
Comment (mmydland) [inline Scripts/ImageFactory/Scripts/MakeGoldenImageVMs.ps1]: $modulePath?

File: Scripts/ImageFactory/Scripts/RetireImages.ps1 [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+$labInfo.Labs = ($labInfo.Labs + $factorylabInfo)
+$labInfo | ConvertTo-Json | Write-Output
+
+# Iterate through all the labs
+foreach ($selectedLab in $labInfo.Labs){
+
+    # Get the list of images in the current lab
+    SelectSubscription $selectedLab.SubscriptionId
+    $AllImages = Get-AzureRmResource -ResourceName $selectedLab.LabName -ResourceGroupName $selectedLab.ResourceGroup -ResourceType 'Microsoft.DevTestLab/labs/customImages' -ApiVersion '2016-05-15'
```
Comment (mmydland) [inline Scripts/ImageFactory/Scripts/RetireImages.ps1]: should AllImages be camelCase

File: Scripts/ImageFactory/Scripts/UpdateBuildAgent.ps1 [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	[Parameter(Mandatory=$true, HelpMessage="The name of the build agent")]
+    [string] $BuildAgent,
+
+	[Parameter(Mandatory=$true, HelpMessage="Either Start or Stop to apply an action to the Virtual Machine")]
+    [string] $Action
+
+)
+
+# find the build agent in the subscription
+$agentVM = Get-AzureRmResource -ResourceGroupName $ResourceGroupName -ResourceType Microsoft.DevTestLab/labs/virtualmachines -ResourceName $DevTestLabName -ApiVersion 2016-05-15 | Where-Object {$_.Name -eq $BuildAgent}
```
Comment (mmydland) [inline Scripts/ImageFactory/Scripts/UpdateBuildAgent.ps1]: parameterize Api-Version

**Pull request merged**